### PR TITLE
Update slim/psr7 and slim/slim dependencies

### PR DESCRIPTION
**Pull request by manavo**
The [contributing guide](https://github.com/slimphp/Slim-Skeleton/blob/master/CONTRIBUTING.md) says to make the PR to the `4.x` branch, but that doesn't seem to exist here.

So just making the PR to `master`, sorry if that's wrong!

### CODE REVIEW

**Comment by manavo** [comment]
Ah, looks like `slim/psr7` requires PHP 7.2 or newer, so the 7.1 test fails.

Happy to update the PHP version requirement in `composer.json` and adjust the Travis tests, if you're happy dropping 7.1!

**Comment by manavo** [comment]
Sorry for many comments, just saw that the latest Slim framework releases require PHP 7.2, so I've gone ahead and updated that too here, since it will then allow projects to be created and actually pull in the latest version of the framework.

I think my PR is now getting a bit out of scope, sorry!

File: composer.json [hunk 1]
```diff
@@ -24,9 +24,9 @@
     "require": {
         "ext-json": "*",
         "monolog/monolog": "^1.24",
-        "php": ">=7.1",
+        "php": "^7.2",
         "php-di/php-di": "^6.0",
-        "slim/psr7": "^0.5",
+        "slim/psr7": "^1.0",
         "slim/slim": "^4.1"
```
Comment (l0gicgate) [inline composer.json:30]: We should update Slim to 4.5 here

Comment (manavo) [reply composer.json]: Updated that too @l0gicgate 👍

**Pull request merged**